github-changelog-generator/github-changelog-generator
# Add Token Location to Verbose Output

## What I did
Fixed issue #1047 where verbose output did not detect when token came from the `CHANGELOG_GITHUB_TOKEN` ENV variable. Updated value to specify if it came from options or ENV vars.
## How I did it
Added new method `censored_token` to `options.rb`. Method censors the token by specifying where the token came from, or if it wasn't used.
## How to verify
- Run the changelog generator with no token in ENV vars or in options. See that verbose output includes: `:token=>"No token used"`
- Run the changelog generator with token in options. See that output includes: `:token=>"Used token from options"`
- Run the changelog generator with token in ENV vars. See that output includes: `:token=>"Used token from environment variable"`
- Run the changelog generator with token in both the ENV vars and options. See that output specifies token is used from options: `:token=>"Used token from options"`
